lighthouse-extension/app/src/popup.js

```diff
@@ -182,15 +146,58 @@ document.addEventListener('DOMContentLoaded', _ => {
         feedbackEl.appendChild(buildReportErrorLink(err));
       }
 
-      stopSpinner();
+      hideRunningSubpage();
       background.console.error(err);
     });
   });
+}
+
+/**
+ * Generates a document fragment containing a list of checkboxes and labels
+ * for the aggregation categories.
+ * @param {!Object<boolean>} selectedAggregations
+ * @param {!Window} background Reference to the extension's background page.
+ * @return {!DocumentFragment}
+ */
+function generateOptionsList(list, selectedAggregations, background) {
+  const frag = document.createDocumentFragment();
+
+  const defaultAggregations = background.getDefaultAggregations();
+  defaultAggregations.forEach(aggregation => {
+    const isChecked = selectedAggregations[aggregation.name];
+    frag.appendChild(createOptionItem(aggregation.name, isChecked));
+  });
+
+  return frag;
+}
```

Comment:
this doesn't use the first `list` argument in here.

I was thinking you could also just have it fetch the background itself (though it doesn't really matter), and since `selectedAggregations` is only used in here (not in the parent), you could even absorb that too:

```js
/**
 * Generates a document fragment containing a list of checkboxes and labels
 * for the aggregation categories.
 * @return {!Promise<!DocumentFragment>}
 */
function generateOptionsList() {
  const frag = document.createDocumentFragment();

  return getBackgroundPage.then(background => {
    return background.loadSelectedAggregations().then(selectedAggregations => {
      const defaultAggregations = background.getDefaultAggregations();
      defaultAggregations.forEach(aggregation => {
        const isChecked = selectedAggregations[aggregation.name];
        frag.appendChild(createOptionItem(aggregation.name, isChecked));
      });
      return frag;
}
```

then below it would be
`generateOptionsList().then(frag => optionsList.appendChild(frag));`